### PR TITLE
Adds Lunchboxes and Premade Lunches (Content)

### PR DESCRIPTION
**Pull request by Optimism333**
## Would the Chief Engineer report to arrivals? He forgot his lunchbox. ##
I did this in a day because I thought it a shame we have food delivery bags but no lunchboxes.

### Lunchboxes! ###
image
- I grabbed a nice sprite that Bay had for an NT lunchbox, and made several others based off it. (Thanks to basil for helping make the trader lunchbox look nicer)
- Currently there are: metal lunchbox, plastic lunchbox, NT lunchbox, Discount lunchbox, Getmore lunchbox, Syndicate lunchbox, Zam lunchbox, Trader lunchbox
- Lunchboxes behave similarly to food delivery bags, but are smaller and can hold less items: 7 instead of 14
- Lunchboxes are either made of metal or plastic (except Discount Dan's lunchbox), and a plain one can be made with a metal sheet or plastic sheet

### Collectibles ###
image
image

Added a collectible lunchbox entry to the merch console. The lunchbox you purchase will be one of 14 unique designs! (Credit to cricket on the Discord for the designs)

### HONK and ... ###
image

The autodrobe stocks 2 clown-themed lunchboxes, and 2 mime-themed lunchboxes (Credit to cricket on the Discord for the designs)

### Prepacked with food! ###
- Some lunchboxes start with food in them. They will have an entree, a side, a sweet, a drink, a condiment, and a spork as a utensil
- The food items are pulled randomly from a list of options, so you never quite know what you're going to get

**NT Lunch Example**
image

**Syndicate Lunch Example**
image

**Discount Lunch Example**
image

**Trader Lunch Example**
image

**Zam Lunch Example**
image

### Where do you get them? ###
- The kitchen vendor now starts with ten NT brand lunchboxes (twice the number of food delivery bags)
- You can make a plain lunchbox with a metal sheet or a plastic sheet
- The cargo merch computer now stocks lunchboxes that have been prepacked with a lunch! They cost 60 credits each
- The cargo merch computer also stocks Zam lunchboxes, prepackaged with a lunch guaranteed to be palatable for a grey. They cost 70 credits each
- The cargo merch computer stocks collectible lunchboxes for 10 credits each (they do not come with food)
- The autodrobe has 2 mime lunchboxes and 2 clown lunchboxes
- The trader outfitter no longer has the two boxes of biscuits, and instead has three (number of trader spawns) lunch boxes prepacked with vox cuisine
- The Getmore Chocolate Corp and Zam Snax vending machines sell empty lunchboxes for 10 credits
- The Discount Dan vending machine sells empty lunchboxes for 5 credits
- The Discount Dan and ZamSnax vending machines stock a very limited supply of prepacked lunchboxes as premium merchandise (only 2 each). They are quite pricey, costing a coin and some credits.
- Syndicate lunchboxes with food are available under "Pointless Badassery" on an uplink for 1 TC

### Some lunchboxes are unique ###
- The Syndicate one is more robust than the rest
- The Discount Dan one is made out of cardboard and utterly useless as a weapon
- The Zam one is acid resistant
- The clown one is a bit slippery

## Feedback ##
Opinions on pricing, implementation, and anything else would be very much appreciated.

[content] [sprites]

:cl:
 * rscadd: Added several themed lunchboxes. Plain ones can be crafted from a metal or plastic sheet
 * rscadd: Added NT brand lunchboxes to the kitchen vendor
 * rscadd: Added two lunchboxes with Discount lunch to the Dan vendor as premium merch. They cost a coin and 30 credits
 * rscadd: Added two lunchboxes with lunch to the Zam vendor as premium merch. They cost a coin and 50 credits
 * rscadd: Added NT lunchboxes with food to the cargo merch computer for 60 credits
 * rscadd: Added Zam lunchboxes with food to the cargo merch computer for 70 credits
 * rscadd: Empty lunchboxes can be purchased from ZamSnax, Getmore, and Discount vending machines
 * rscadd: Added a pre-filled Syndicate lunchbox to "Pointless Badassery" for 1 TC
 * rscadd: Added collectible lunchboxes to the cargo merch computer
 * rscadd: Added mime and clown themed lunchboxes to the autodrobe
 * tweak: Replaced the two biscuit boxes in the trader outfitter with three prepacked lunchboxes

### CODE REVIEW

**Comment by Eneocho** [comment]
very cool
would like to see a premium lunchbox (maybe on the merchvendor for a lot of moneys) that has rare foods, like the potent ham, dangerously cheesy poutine, clown's tears/roboticist's tears, etc

**Comment by Eneocho** [comment]
Also, would you consider adding the gunk foods to the discount lunchbox? I feel it'd fit, since, if they can sell a cockroach stuffed between two buns, why sell regular burgers?

**Comment by SonixApache** [comment]
Make sure the syndie toolbox contains syndicakes and donk pockets. There's a table in the nuke ops shuttle that has cakes, throw the lunchbox on there. Slap it in the uplink for 1tc - agents can't fight on an empty stomach.

**Comment by DorainGrey** [comment]
It'd be cool if the ERT also get the lunchboxes in their starting loadout.
Centcomm sent them to the station with a packed lunch.

**Comment by Blithering** [comment]
Would it be possible to add a bias towards sandwich type dishes in the lunchbox?

**Comment by Optimism333** [comment]
> Make sure the syndie toolbox contains syndicakes and donk pockets. There's a table in the nuke ops shuttle that has cakes, throw the lunchbox on there. Slap it in the uplink for 1tc - agents can't fight on an empty stomach.

Sure, I'll put a filled Syndie lunchbox on the uplink for 1 TC. Sabotaging stations is hard work, after all.

**Comment by Optimism333** [comment]
> Would it be possible to add a bias towards sandwich type dishes in the lunchbox?

Maybe, but I think it's okay as it is. Of the 24 options for an entree, 8 are a sandwich or burger, so there's about a 1 in 3 chance of getting a sandwich type dish.

**Comment by Optimism333** [comment]
Lunchboxes can now be quickly filled and emptied of food items, just like food delivery bags. A nice little bit of qol.

https://github.com/vgstation-coders/vgstation13/assets/81007356/fde03880-f9ff-4177-ab50-70ed65623d1d

**Comment by Optimism333** [comment]
Adds a prepackaged Syndicate lunchbox to the uplink under "Pointless Badassery" for 1 TC. Eat your lunch while you make sure Nanotrasen scum don't get to enjoy theirs.

Syndicate lunches are superior to NT ones in a few ways.
- Soups and salads that are main dishes in NT lunches are side dishes in Syndicate lunches. You might get a sandwich AND soup. Good god...
- You get a whole bottle of condiment, rather than a tiny packet.
- Generally the entree, side, and sweet choices are more exotic and/or nutritious.
- Drink might be a Gunka Cola or another useful premium drink.

Examples:
image

image

**Comment by Optimism333** [comment]
Slightly tweaked lunchboxes. They are now one weight class bigger, so they can't fit in pockets or boxes.

They have one extra storage slot: 6 -> 7

Snapshots in the pr have been updated to reflect this.

**Comment by NigglyWiggly** [comment]
Bento boxes when

**Comment by Optimism333** [comment]
> Bento boxes when

I'll happily add it if someone else would be kind enough to make the sprite. I tried a few times, it didn't come out well.

**Comment by Optimism333** [comment]
image
image

Added a lot more lunchbox designs! (Credit to cricket in the Discord for all of these)

There is a new entry under the cargo merch computer called "Collectible Lunchboxes". When you buy one, you will get a lunchbox with a unique design (14 random options).

image

Lunchboxes are now split into plastic and metal types: 
- Zam, Trader, and Syndie lunchboxes are metal.
- NT, Getmore, and Collectible lunchboxes are plastic.
- You can build a plain lunchbox from a metal sheet OR a plastic sheet

image

Added lunchboxes with clown and mime themed decor! Two of each can be found in the autodrobe

The clown one is more slippery than the other lunchboxes

File: code/game/objects/items/weapons/storage/lunchbox.dm [hunk 1]
```diff
@@ -359,96 +349,253 @@
...
+		if(4)
+			name = "Carp lunchbox"
+			desc = "A little plastic lunchbox. This one has a picture of a space carp on the side."
+			icon_state = "lunchbox_carp"
+			item_state = "toolbox_purple"
+		if(5)
+			name = "Carp lunchbox"
+			desc = "A little plastic lunchbox. This one has a picture of a space carp on the side."
+			icon_state = "lunchbox_carp"
+			item_state = "toolbox_purple"
```
Comment (Eneocho) [inline code/game/objects/items/weapons/storage/lunchbox.dm]: Duplicate carp lunchbox

Comment (Eneocho) [reply code/game/objects/items/weapons/storage/lunchbox.dm]: Same thing with the jannie lunchbox

Comment (Optimism333) [reply code/game/objects/items/weapons/storage/lunchbox.dm]: Thanks for catching that. Should be fixed now.

**Comment by Eneocho** [comment]
Added to the vending machine wiki page

**Pull request merged**